sql/src/main/java/io/crate/operation/reference/sys/check/node/DiskWatermarkNodesSysCheck.java

```diff
@@ -0,0 +1,102 @@
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.check.node;
+
+import io.crate.metadata.settings.CrateSettings;
+import io.crate.metadata.settings.StringSetting;
+import io.crate.operation.reference.sys.check.AbstractSysNodeCheck;
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.RatioValue;
+import org.elasticsearch.monitor.fs.FsInfo;
+import org.elasticsearch.monitor.fs.FsProbe;
+
+import java.io.IOException;
+
+
+abstract class DiskWatermarkNodesSysCheck extends AbstractSysNodeCheck {
+
+    private static final ESLogger LOGGER = Loggers.getLogger(DiskWatermarkNodesSysCheck.class);
+
+    private final StringSetting watermarkSetting;
+    private final Settings settings;
+    private final FsProbe fsProbe;
+
+    DiskWatermarkNodesSysCheck(int id,
+                               String description,
+                               StringSetting watermarkSetting,
+                               Severity severity,
+                               ClusterService clusterService,
+                               Settings settings,
+                               FsProbe fsProbe) {
+        super(id, description, severity, clusterService);
+        this.watermarkSetting = watermarkSetting;
+        this.settings = settings;
+        this.fsProbe = fsProbe;
+    }
+
+    @Override
+    public boolean validate() {
+        try {
+            return !thresholdEnabled() ||
+                    validate(fsProbe.stats(), watermarkSettingInPercents(), watermarkSettingInBytes());
+        } catch (IOException e) {
+            LOGGER.error("Unable to determine the node disk usage while validating high/low disk watermark check: ", e);
+            return false;
+        }
+    }
+
+    protected abstract boolean validate(FsInfo fsInfo, double diskWatermarkPercents, long diskWatermarkBytes);
+
+    private double watermarkSettingInPercents() {
+        return thresholdPercentageFromWatermark(watermarkSetting.extract(settings));
```

Comment:
afaik there is no other usage of the called method, so why wrap it in another method?